students/laura_denney/lesson09/mailroom_objectoriented.py

```diff
@@ -0,0 +1,280 @@
+#   Laura Denney, 12/7/18, Started Object Oriented Mail Room
+#   Laura Denney, 12/17/18, Added Saving / Loading Functionality
+#-------------------------------------------------#
+
+#adding date to txt file
+from datetime import date
+
+#file handling help
+import os
+
+#Donor Class contains name, donations
+class Donor(object):
+    name = ""
+    donations = None
+
+    @property
+    def sum_donations(self):
+        return sum(self.donations)
+    @property
+    def total_donations(self):
+        return len(self.donations)
+    @property
+    def average_donation(self):
+        return self.sum_donations / self.total_donations
+
+    def __init__(self, name, *args):
+        self.name = name.lower()
+        self.donations = []
+        if args:
+            for arg in args:
+                self.donations.append(arg)
+
+    def __repr__(self):
+        repr_str = "Donor('{}', " + ", ".join(["{}"] * len(self.donations)) + ")"
+        return repr_str.format(self.name, *self.donations)
+
+    def __eq__(self, other):
+        return ((self.sum_donations, self.name) == \
+                (other.sum_donations, other.name))
+
+    def __lt__(self, other):
+        return ((self.sum_donations, self.name) < \
+                (other.sum_donations, other.name))
+
+    def append(self, donation):
+        self.donations.append(donation)
+####################################################
+#Donors Class - manipulates a list of donors
+class Donors(object):
+    donor_list = None
+
+    def __init__(self, *args):
+        self.donor_list = []
+        for arg in args:
+            self.donor_list.append(arg)
+
+    def __repr__(self):
+        repr_str = "Donors(" + ", ".join(["{}"] * len(self)) + ")"
+        return repr_str.format(*self.donor_list)
+
+    def __len__(self):
+        return len(self.donor_list)
+
+    def add_new_donor(self, donor):
+        self.donor_list.append(donor)
+
+
+    def is_current_donor(self, donor):
+        return donor.lower() in repr(self)
+
+    def get_list(self):
+        names = [donor.name.title() for donor in self.donor_list]
+        strformat = "\nWe have {} current donors: " + ", ".join(["{}"] * len(names))
+        return strformat.format(len(names), *names)
+
+    def create_string_report(self):
+        self.donor_list.sort(reverse = True)
+        report = '''
+Donor Name                | Total Given | Num Gifts | Average Gift
+------------------------------------------------------------------'''
+        strformat = '\n{:<26}${:>13.2f}{:^12}${:>13.2f}'
+        for donor in self.donor_list:
+            report +=  strformat.format(donor.name.title(),donor.sum_donations,
+                                    donor.total_donations, donor.average_donation)
+        return report
+
+    def get_donor(self, name):
+        for donor in self.donor_list:
+            if donor.name == name:
+                return donor
+
+    def check_current_donor(self, name, donation):
```

Comment:
I think this function does a bit more than "checks" - isn't it updating donors data? Should this be called `update_donors` ?